k2/csrc/cuda/ops.h

```diff
@@ -1,71 +1,149 @@
 // k2/csrc/cuda/ops.h
 
-// Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey)
+// Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey
+//                                                   Haowen Qiu)
 
 // See ../../LICENSE for clarification regarding multiple authors
 
 #ifndef K2_CSRC_CUDA_OPS_H_
 #define K2_CSRC_CUDA_OPS_H_
 
-#include "k2/csrc/cuda/array.h"
-
+#include <cassert>
+#include <type_traits>
 
-// Note, I'm not sure about the name of this file, they are not ops like in TensorFlow, but
-// procedures..
+#include "k2/csrc/cuda/array.h"
+#include "k2/csrc/cuda/context.h"
+
+// Note, I'm not sure about the name of this file, they are not ops like in
+// TensorFlow, but procedures..
+
+// TODO(haowen): this function may be moved to some header files for log (e.g.
+// `log.h`)
+inline cudaError_t CheckCuda(cudaError_t status) {
+#if defined(DEBUG)
+  if (status != cudaSuccess) {
+    fprintf(stderr, "CUDA Runtime Error: %s\n", cudaGetErrorString(status));
+    assert(status == cudaSuccess);
+  }
+#endif
+  return status;
+}
+
+namespace {
+// TODO(haowen): manage/load block config with some classes? then we can get
+// different configuration depending on num_elements and data type.
+// block size for matrix transpose.
+constexpr int TRANS_TILE_DIM = 32;
```

Comment:
`UPPERCASE_NAME` is usually for macros, while `kConstName` is for constants.

Reply:
Fixed